circuits/plonk/src/constraints.rs

```diff
@@ -65,16 +66,17 @@ pub struct ConstraintSystem<F: FftField>
     pub ps8:    Evaluations<F, D<F>>,       // poseidon selector over domain.d8
 
     // ECC arithmetic selector polynomials
-    pub addl3:  Evaluations<F, D<F>>,       // EC point addition selector evaluations w over domain.d4
-    pub addl4:  Evaluations<F, D<F>>,       // EC point addition selector evaluations w over domain.d8
+    pub addl4:  Evaluations<F, D<F>>,       // EC point addition selector evaluations w over domain.d4
     pub mul1l:  Evaluations<F, D<F>>,       // scalar multiplication selector evaluations over domain.d4
     pub mul2l:  Evaluations<F, D<F>>,       // scalar multiplication selector evaluations over domain.d8
     pub emul1l: Evaluations<F, D<F>>,       // endoscalar multiplication selector evaluations over domain.d4
     pub emul2l: Evaluations<F, D<F>>,       // endoscalar multiplication selector evaluations over domain.d4
     pub emul3l: Evaluations<F, D<F>>,       // endoscalar multiplication selector evaluations over domain.d8
 
-    pub l0:     Evaluations<F, D<F>>,       // 0-th Lagrange evaluated over domain.d8
+    pub l04:    Evaluations<F, D<F>>,       // 0-th Lagrange evaluated over domain.d4
```

Comment:
no, it wouldn't help memory if we avoided holding onto this separately. It would make the use of memory/performance worse with allocating it on stack every rime used.

Reply:
I was suggesting we just use the subset of l08 and not reallocate a new vector. It would admittedly be awkward because we would have to use special operations rather than the ones implemented for the `Evaluations` struct.